platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r85/ProblemProgressEventCrossVersionTest.groovy

```diff
@@ -17,34 +17,50 @@
 package org.gradle.integtests.tooling.r85
 
 import groovy.json.JsonSlurper
-import org.gradle.api.problems.Severity
-import org.gradle.integtests.tooling.fixture.TargetGradleVersion
 import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
 import org.gradle.integtests.tooling.fixture.ToolingApiVersion
 import org.gradle.tooling.BuildException
 import org.gradle.tooling.events.ProgressEvent
 import org.gradle.tooling.events.ProgressListener
-import org.gradle.tooling.events.problems.ProblemDescriptor
+import org.gradle.tooling.events.problems.BaseProblemDescriptor
 import org.gradle.tooling.events.problems.ProblemEvent
 
 @ToolingApiVersion("=8.5")
-@TargetGradleVersion("=8.5")
 class ProblemProgressEventCrossVersionTest extends ToolingApiSpecification {
 
-    class MyProgressListener implements ProgressListener {
-        List<ProblemDescriptor> allProblems = []
-
-        @Override
-        void statusChanged(ProgressEvent event) {
-            if (event instanceof ProblemEvent) {
-                this.allProblems.addAll(event.getDescriptor())
+    def "test failure context"() {
```

Comment:
❌ The two tests in this class seems identical in terms of covered scenarios as well as the method bodies. The only difference is this bit:
```
problems.every {
            it == [:]
        }
```

Can we merge the the two test methods?